### PR TITLE
Fix handling the lock file

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/15)

### CODE REVIEW

File: src/vltrace.c [hunk 1]
```diff
@@ -261,42 +250,27 @@ try_to_read_lock_file()
...
-	} else {
-		perror("fopen");
-		WARNING("error opening lock file for writing");
+	if (flock(FD_lock, LOCK_EX | LOCK_NB)) {
+		/* cannot grab the file lock, try to read the PID from file */
+		return try_to_read_lock_file();
+	}
+
+	/* write PID to the lock file */
+	(void) asprintf(&buf, "%i \n", getpid());
```
Comment (marcinslusarz) [inline src/vltrace.c]: I suspect I know why you put space before \n, but this is ugly approach - just truncate the file before writing anything.

**Review by marcinslusarz — CHANGES_REQUESTED** [review]

File: src/vltrace.c (hunk 1 above)
Comment (ldorau) [reply src/vltrace.c]: Done

**Comment by marcinslusarz** [comment]
*[src/vltrace.c, line 74 at r2](https://reviewable.io:443/reviews/pmem/vltrace/15#-Kq2wZDpxboru4c1pfAQ:-Kq2wZDpxboru4c1pfAR:btjogec) ([raw file](https://github.com/pmem/vltrace/blob/91f20cc8de65c3fb39f1b239f8618bccec7bca64/src/vltrace.c#L74)):*
> ```C
> #define VLTRACE_NAME		"vltrace"
> #define LEN_VLTRACE_NAME	7
> #define FILE_LOCK		"/tmp/"VLTRACE_NAME"-lock-file.pid"
> ```

This seem like an unrelated change (separate commit or at least mention it in commit message please) and I'm not sure it's correct.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/15)*

**Comment by ldorau** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/vltrace.c, line 74 at r2](https://reviewable.io:443/reviews/pmem/vltrace/15#-Kq2wZDpxboru4c1pfAQ:-Kq3OLX4oVlsYJvmavP3:b-910xan) ([raw file](https://github.com/pmem/vltrace/blob/91f20cc8de65c3fb39f1b239f8618bccec7bca64/src/vltrace.c#L74)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

This seem like an unrelated change (separate commit or at least mention it in commit message please) and I'm not sure it's correct.
</blockquote></details>

Path set to /var/lock

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/15)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/15#-:-Kq7LZJjafnDcwkyBm_o:bnfp4nl)*

**Review by marcinslusarz — APPROVED** [review]

**Pull request merged**